### ISSUE TITLE
about how to calculate the AUC score of the model testing on the testdataset

### ISSUE THREAD

**Issue by algorithmlixuan** [polimi-ispl/icpr2020dfdc#17]
hi thanks for the great work and thanks for answer my issues 16. I read your test_model.py code, but I didn't find the code about how to calculate the AUC score of the model testing on the testdataset. So I write it myself, I download your pretrained Xception_FFPP(best_val.pth). I run the script:
`python test_model.py 
--model_path /media/lixuan/others/dataset/icpr2020dfdc/weights/binclass/net-Xception_traindb-ff-c23-720-140-140_face-scale_size-64_seed-41/bestval.pth 
--testsets ff-c23-720-140-140 dfdc-35-5-10 
--dfdc_faces_df_path $DFDC_FACES_DF 
--dfdc_faces_dir $DFDC_FACES_DIR 
--ffpp_faces_df_path $FFPP_FACES_DF 
--ffpp_faces_dir $FFPP_FACES_DIR 
--device $DEVICE 
--override`
and get the result,
then I write code to calculate the AUC score like this:

```
import pickle
import pandas as pd
import numpy as np
from sklearn.metrics import roc_auc_score
import torch
res_path = 'results/net-Xception_traindb-ff-c23-720-140-140_face-scale_size-64_seed-41_bestval/ff-c23-720-140-140_test.pkl'
with open(res_path, 'rb') as res:
    result = pickle.load(res)
labels = result['label'].values
labels_bool2value = []
for i in range(labels.shape[0]):
    if labels[i]:
        labels_bool2value.append(0)
    else:
        labels_bool2value.append(1)
pred = torch.sigmoid(torch.from_numpy(result['score'].values)).numpy()
auc = roc_auc_score(np.array(labels_bool2value), pred)
print(auc)
```

but the AUC score I get is 0.46364417251275514.It's lower than the result provided in your paper.Would you like to tell me what is wrong with my code ? Or would you like to share your code about how to calculate the AUC score of the model testing on the testdataset. thank you very much

**Comment by nicobonne** [comment]
Hi @algorithmlixuan, have you checked [Analyze results notebook](https://github.com/polimi-ispl/icpr2020dfdc/blob/master/notebook/Analyze%20results.ipynb) out? It contains the way we used to compute the AUC.

**Comment by CrohnEngineer** [comment]
Hey @algorithmlixuan ,

two other comments in addition to the pointer given by @nicobonne : 
1.  in the snippet of code you provided, you wrote `if labels[i]:  labels[i] = 0`, but in the DataFrame the label `True `corresponds to a FAKE video, not a REAL one. The networks are trained so that for a FAKE video they return a sigmoid score close to 1, so you are swapping the labels between the two classes when computing the AUC score (you should have something around 1-0.46=0.54);
2. from the path where you stored the test results (`results/net-Xception_traindb-ff-c23-720-140-140_face-scale_size-64_seed-41_bestval`), I see that the network has been trained on faces of size 64x64 pixels: are you sure this is one of the models we provided? We trained with faces of size 224x224 pixels, so I am a bit confused on what is happening here. Keep in mind that in order to compare with our results, you should have a network trained and tested on that input size.

Let us know if these suggestions help.
Bests,

Edoardo

**Comment by algorithmlixuan** [comment]
> Hi @algorithmlixuan, have you checked [Analyze results notebook](https://github.com/polimi-ispl/icpr2020dfdc/blob/master/notebook/Analyze%20results.ipynb) out? It contains the way we used to compute the AUC.

thanks. I ignored the file before. your code help me a lot  . And I will try to train with faces of size 224x224 pixels and find if I can get the result in your paper.thanks again

**Comment by algorithmlixuan** [comment]
> Hey @algorithmlixuan ,
> 
> two other comments in addition to the pointer given by @nicobonne :
> 
> 1. in the snippet of code you provided, you wrote `if labels[i]:  labels[i] = 0`, but in the DataFrame the label `True `corresponds to a FAKE video, not a REAL one. The networks are trained so that for a FAKE video they return a sigmoid score close to 1, so you are swapping the labels between the two classes when computing the AUC score (you should have something around 1-0.46=0.54);
> 2. from the path where you stored the test results (`results/net-Xception_traindb-ff-c23-720-140-140_face-scale_size-64_seed-41_bestval`), I see that the network has been trained on faces of size 64x64 pixels: are you sure this is one of the models we provided? We trained with faces of size 224x224 pixels, so I am a bit confused on what is happening here. Keep in mind that in order to compare with our results, you should have a network trained and tested on that input size.
> 
> Let us know if these suggestions help.
> Bests,
> 
> Edoardo

thanks, I modified my code as you say.it gets the result the same as your notebook. And I will try to train with faces of size 224x224 pixels and find if I can get the result in your paper.thanks again
